### DISCUSSION TITLE
Generic Serde for kotlinx.serialization?

### DISCUSSION THREAD

**Discussion by manosbatsis** [yahoo/elide#2738] [General]
I'm using kotlin multiplatform to share code between an elide/spring back-end and mobile/desktop/web client apps. This forces me to use e.g. [kotlinx-datetime](https://github.com/Kotlin/kotlinx-datetime) instead of Java 8 Date/Time, [KMM UUID](https://github.com/benasher44/uuid) instead of java.util.UUID and so on.

Creating a Serde (and JPA AttributeConverter) for these doesn't feel right since they support [kotlinx.serialization](https://github.com/Kotlin/kotlinx.serialization) out of the box, so I was wondering if anyone out there has a more generic approach. 

Alternatively i might as well create a public repo with what i have or ideally contribute to elide.

**Comment by aklish** [comment]
I'm having a hard time imaging a generic solution here (but open to suggestions).  If you feel like Elide might be a good home for your Serdes, we will definitely considering adding them.

**Comment by manosbatsis** [comment]
Created a [basic repo](https://github.com/manosbatsis/kmp-converters) that seems to work for us in a project with Elide and Spring Boot but  is largely untested/crude. Modules are supposed to provide JPA and Serde converters with some flexibility, i.e. plain JPA, Spring Boot, Elide Standalone and Elide with Spring Boot.

It's still very alpha but any feedback is welcome. If you're interested in taking over the Elide modules i'd be happy to prepare and transfer a repo to you.

In the mean time i'll try to add tests and polish.
